airflow/models.py

```diff
@@ -1793,11 +1793,12 @@ def _set_relatives(self, task_or_task_list, upstream=False):
             if not isinstance(task, BaseOperator):
                 raise AirflowException('Expecting a task')
             if upstream:
-                self.append_only_new(task._downstream_list, self)
+                task.append_only_new(task._downstream_list, self)
                 self.append_only_new(self._upstream_list, task)
             else:
-                self.append_only_new(task._upstream_list, self)
                 self.append_only_new(self._downstream_list, task)
+                task.append_only_new(task._upstream_list, self)
+
```

Comment:
Simply changing the order and the task on which we call `append_only_new` here, to make sure the first error in `append_only_new` outputs tasks names from upstream to downstream in the message `Dependency {self}, {item} already registered` when there's an error.